evals/runners/run_retrieval.py

```
@@ -28,12 +42,33 @@
+def _agent_preferred_implants(agent_name: str | None) -> tuple[str, ...]:
+    """Return the declared preferred_implants of an agent (as stems, no .mdc).
+
+    Cached per-agent so a 110-row eval triggers at most one frontmatter read
+    per distinct ``expected_agent``. Missing agent / missing field → empty
+    tuple; callers treat empty as "no derivation possible for this sample".
+    """
+    if not agent_name:
+        return ()
+    try:
+        meta = get_agent_metadata(agent_name) or {}
+    except Exception:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)